index.test.js

```diff
@@ -0,0 +1,283 @@
+'use strict';
+
+/**
+ * Module dependencies.
+ */
+
+const { AssertionFailedError, ValidationFailedError } = require('@uphold/http-errors');
+const createValidator = require('.');
+const asserts = require('validator.js-asserts');
+const _ = require('lodash');
+
+/**
+ * Extra asserts
+ */
+
+function AlwaysValid() {
+  // eslint-disable-next-line no-underscore-dangle
+  this.__class__ = 'AlwaysValid';
+
+  this.validate = function() {
+    return true;
+  };
+
+  return this;
+}
+
+function AlwaysInvalid() {
+  // eslint-disable-next-line no-underscore-dangle
+  this.__class__ = 'AlwaysInvalid';
+
+  this.validate = function() {
+    return false;
+  };
+
+  return this;
+}
+
+/**
+ * Test `Validator`.
+ */
+
+describe('Validator', () => {
+  it('should expose `assert`, `is` and `validate` functions', () => {
+    const { assert, is, validate } = createValidator({
+      AssertionError: AssertionFailedError,
+      ValidationError: ValidationFailedError
+    });
+
+    expect(assert).toBeInstanceOf(Function);
+    expect(is).toBeInstanceOf(Function);
+    expect(validate).toBeInstanceOf(Function);
+  });
+
+  it('should expose custom asserts in pascal case', () => {
+    const { is } = createValidator({
+      AssertionError: AssertionFailedError,
+      ValidationError: ValidationFailedError
+    });
+
+    for (const assert in asserts) {
+      expect(is.prototype).toHaveProperty(assert);
+    }
+  });
+
+  it('should expose extra custom asserts in pascal case', () => {
+    const { is } = createValidator({
+      AssertionError: AssertionFailedError,
+      ValidationError: ValidationFailedError,
+      extraAsserts: { AlwaysInvalid, AlwaysValid }
+    });
+
+    for (const assert in asserts) {
+      expect(is.prototype).toHaveProperty(assert);
+    }
```

Comment:
This one can be removed, is already being tested on the above test right?

Reply:
Yep, my bad. ✅